inc/switchtec/switchtec.h

```diff
@@ -355,8 +355,21 @@ switchtec_boot_phase(struct switchtec_dev *dev);
 int switchtec_set_pax_id(struct switchtec_dev *dev, int pax_id);
 int switchtec_echo(struct switchtec_dev *dev, uint32_t input, uint32_t *output);
 int switchtec_hard_reset(struct switchtec_dev *dev);
+
+typedef struct {
+     uint8_t  : 3;
+     uint8_t minor_state : 4;
+     uint8_t major_state : 4;
+     uint8_t  : 2;
+     uint8_t link_rate: 2;
+     uint32_t : 17;
+     uint32_t tstamp: 26;
+     uint8_t : 6;
+}__attribute__((packed)) ltssm_log_data;
```

Comment:
Bit-fields definition is vague as different compiler/machine may have different interpretations of the ordering of the bits. Normally we use shifting and masking to get proper value of some bits.

Reply:
Is there an example in the code base I can use as a template?

Reply:
Yes, never use bitfields for anything where you care where the bits actually are.